lib/request-handler.js

```diff
@@ -181,6 +187,18 @@ function handleRequest(cacheHeaderConfig, handlerOptions) {
         }
         callback(err, res, body)
       })
+      r.on('data', chunk => {
+        buffer += chunk
+        if (buffer.length > config.public.maxResponseSizeBytes) {
```

Comment:
Since AFAICT we don't care about the actual value of the `buffer`, could you accumulate the length instead of the contents to save memory?